pkgs/top-level/module/module.nix

```
@@ -0,0 +1,176 @@
+  be imported in a submodule.
+
+  This module is based on the NixOS `misc/nixpkgs.nix` module, but it gets rid
+  of legacy options. It may be used to replace that module after legacy options
+  have been deprecated and removed.
+
+  As this module declares an option named `config`, you're recommended to pass
+  `shorthandOnlyDefinesConfig = true` to `types.submoduleWith`.
+
+  Maintainer: @roberth
+
+  module: https://nixos.org/manual/nixpkgs/unstable/#module-system
+*/
+{ lib, config, options, ... }:
+
+let
+  inherit (lib) mkOption types literalExpression;
+
+  optDefaultPrio = (lib.mkOptionDefault null).priority;
+
+  isCross = config.buildPlatform != config.hostPlatform;
+  systemArgs =
+    if isCross
+    then {
+      localSystem = config.buildPlatform;
+      crossSystem = config.hostPlatform;
+    }
+    else {
+      localSystem = config.hostPlatform;
+    };
+
+  pkgs =
+      import ../default.nix (
+        # NOTE: make inputsSet covers all parameters; see _optimize
+        #       Convenience options that write to inputsSet should be omitted.
+        #       This point is that inputsSet must form a unique key so that
+        #       memoization optimizations are safe.
+        {
+          inherit (config) config overlays;
+        } // systemArgs
+      );
+
+  # TODO: write a better type that checks that the definition ordering is
+  #       unambiguous under import order.
+  overlayType = types.functionTo (types.functionTo types.raw);
+
+  # A unique key for the Nixpkgs package set to be instantiated.
+  #
+  # This is only relevant for _optimize - ie when you're writing or troubleshooting
+  # a module system application that uses _optimize to memoize things.
+  #
+  # Default values and unset values are omitted so that future additions to the
+  # key are usable with _optimize implementations that are not aware of them,
+  # as long as these additions are unused.
+  inputsSet = lib.filterAttrs
+    (_: v: v != v.isDefined && v.highestPrio < optDefaultPrio)
+    {
+      inherit (options) hostPlatform buildPlatform overlays config;
+    };
+
+in
+{
+  options = {
+    hostPlatform = mkOption {
+      type = types.coercedTo types.str lib.systems.elaborate types.attrs;
+      description = lib.mdDoc ''
+        Specifies the platform where the package will run.
+
+        To cross-compile, also set the `buildPlatform` option.
+
+        Ignored when the `pkgs` option is overridden.
+      '';
+    };
+    buildPlatform = mkOption {
+      type = types.coercedTo types.str lib.systems.elaborate types.attrs;
+      description = lib.mdDoc ''
+        Specifies the platform where the package will be built. This is
+        optional; by default Nixpkgs will perform native compilation.
+
+        By default, packages are built on the system where it runs, but
+        you can change where they're built. Setting this option will
+        cause NixOS to be cross-compiled.
```

```suggestion
        Setting this option will cause NixOS to be cross-compiled.
```

The first sentence is redundant with the one before.